### PR TITLE
Improve javadoc for some addons

### PR DESCRIPTION
**Pull request by holgerfriedrich**
Follow-up of #15667.

Fixes javadoc issues not yet handled in #15667:
* "invalid html" (due to special characters in javadoc)
* invalid tags
* some broken links
* introduce custom tags @ implNote and @ apiNote on top level (previously only available in knx addon)

### CODE REVIEW

File: bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/dto/HeatingModel.java
Comment (lsiepel) [inline bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/dto/HeatingModel.java:226]: Just curiosity why should < be changed to html antities and  > not ?

**Review by hmerk — APPROVED** [review]
LGTM for ekey, intesis and wemo

File: bundles/org.openhab.binding.energidataservice/src/main/java/org/openhab/binding/energidataservice/internal/api/DatahubTariffFilter.java
Comment (jlaur) [inline bundles/org.openhab.binding.energidataservice/src/main/java/org/openhab/binding/energidataservice/internal/api/DatahubTariffFilter.java:21]: This was already addressed in #15651 where the link was removed. The name of the dataset is DatahubPricelist while the name of the corresponding DTO is DatahubPricelistRecords. I would therefore prefer to have this change reverted, and probably you need to rebase as well.

File: bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/dto/HeatingModel.java
Comment (holgerfriedrich) [reply bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/dto/HeatingModel.java:226]: < is the open tag marker so html will look for a known tag after this character.... no need to escape >

File: bundles/org.openhab.binding.energidataservice/src/main/java/org/openhab/binding/energidataservice/internal/api/DatahubTariffFilter.java
Comment (holgerfriedrich) [reply bundles/org.openhab.binding.energidataservice/src/main/java/org/openhab/binding/energidataservice/internal/api/DatahubTariffFilter.java:21]: I rebased and got it wrong.... I will revert.

**Review by marcelrv — APPROVED** [review]
LGTM max,windcentrale

**Review by theater — APPROVED** [review]

**Review by MrRonfo — APPROVED** [review]
LGTM for bticinosmarther binding, thanks

**Review by andrewfg — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/dto/clip2/helper/Setters.java
Comment (holgerfriedrich) [reply bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/dto/clip2/helper/Setters.java:84]: I added this as requested. I think we cannot do this for all of the types mentioned in javadoc. The PR just touches the ones containing a < character (causes an error "invalid html").

**Comment by daMihe** [comment]
LGTM for icalendar

**Review by morph166955 — APPROVED** [review]
LGTM for neeo

**Review by andrewfg — APPROVED** [review]

File: bundles/org.openhab.binding.pilight/src/main/java/org/openhab/binding/pilight/internal/dto/Code.java
Comment (stefanroellin) [inline bundles/org.openhab.binding.pilight/src/main/java/org/openhab/binding/pilight/internal/dto/Code.java:1]: Replace link with https://manual.pilight.org/development/socket/index.html since the old one does not work anymore.

File: bundles/org.openhab.binding.pilight/src/main/java/org/openhab/binding/pilight/internal/dto/Config.java
Comment (stefanroellin) [inline bundles/org.openhab.binding.pilight/src/main/java/org/openhab/binding/pilight/internal/dto/Config.java:1]: Replace link with https://manual.pilight.org/development/socket/index.html since the old one does not work anymore.

File: bundles/org.openhab.binding.pilight/src/main/java/org/openhab/binding/pilight/internal/dto/Version.java
Comment (stefanroellin) [inline bundles/org.openhab.binding.pilight/src/main/java/org/openhab/binding/pilight/internal/dto/Version.java:1]: Replace link with https://manual.pilight.org/development/socket/index.html since the old one does not work anymore.

**Review by stefanroellin — CHANGES_REQUESTED** [review]
Links in the Pilight binding are no longer valid. Can you replace them with 
https://manual.pilight.org/development/socket/index.html

File: bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/BoschSHCHandler.java
Comment (david-pace) [inline bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/BoschSHCHandler.java:117]: There are two issues with this link:
* There are 3 overloaded variants of the method `registerService()`, so the link needs to be qualified. The correct link would look like this: `{@link #registerService(BoschSHCService, Consumer, Collection, boolean)}` (without the generic type arguments)
* I noticed that the clients are not supposed to register their services in `initialize()`, but in the more specific method `initializeServices()`. So the Javadoc is actually misplaced here. So the best solution would be to replace the Javadoc of `initializeServices()` with the sentence `Use this method...` from this Javadoc. I would understand if this is out of scope for this pull request, so you can just fix the link and we can move the Javadoc later if you prefer.

**Review by david-pace — CHANGES_REQUESTED** [review]
Found one invalid Javadoc link

File: bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/DaikinBindingConstants.java
Comment (jlaur) [inline bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/DaikinBindingConstants.java:28]:
```suggestion
 * @author <a href="mailto:tim@timwaterhouse.com">Tim Waterhouse</a> - Initial contribution
 * @author <a href="mailto:paul@smedley.id.au">Paul Smedley</a> - Modifications to support Airbase Controllers
```

Would you prefer that I comment all to keep track of them?

Alternative would be to drop the e-mail address entirely, but since it's already there and we are still using the `@author` tag, I would probably favor my proposal above. Ultimately I think we should completely stop using the `@author` tag and leave this to Git. @openhab/add-ons-maintainers - for sharing my opinion. 😉

Comment (holgerfriedrich) [reply bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/DaikinBindingConstants.java:28]: My preference would be to remove the mail addresses. This would match the may we are doing it in other source files. I don't need comments for each occurrence, just give me some time do to it....

Comment (holgerfriedrich) [reply bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/DaikinBindingConstants.java]: If I have not overlooked some, I have removed all email addresses from @author tags.
Hope  everybody is fine with that. But it should be fine, credit to the author is still given.

File: bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/BoschSHCHandler.java
Comment (holgerfriedrich) [reply bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/BoschSHCHandler.java]: Thanks! I tried do do as requested, could you pls check?

File: bundles/org.openhab.binding.pilight/src/main/java/org/openhab/binding/pilight/internal/dto/Code.java
Comment (holgerfriedrich) [reply bundles/org.openhab.binding.pilight/src/main/java/org/openhab/binding/pilight/internal/dto/Code.java:1]: done, thanks for providing the new link.

File: bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/BoschSHCHandler.java
Comment (david-pace) [reply bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/BoschSHCHandler.java]: Perfect, thanks 😎

**Review by david-pace — APPROVED** [review]
LGTM for boschshc, thanks 👍

**Review by Jamstah — APPROVED** [review]

**Review by jlaur — APPROVED** [review]
Thanks for taking care! 👍

**Pull request merged**
